### ISSUE TITLE
Gameboy: Allow importing MBC2 save files from other emulators

### ISSUE THREAD

**Issue by RetroEdit** [mgba-emu/mgba#1743]
A user posted on Discord about their save file from VBA not loading properly in mGBA. Upon further investigation, said save file loaded in BGB (1.5.7) and Sameboy (v0.12.3), so it seems like this is possibly a bug with something in mGBA.

The save file is attached. Upon loading, it's supposed to be selected on the "Continue" screen and then advance to a save file with the name 'Sumo'. On mGBA, it behaves like no save file exists (but it does load save files it created fine).

[Final_Fantasy_Adventure_Seiken_Densetsu_USA.zip](https://github.com/mgba-emu/mgba/files/4548456/Final_Fantasy_Adventure_Seiken_Densetsu_USA.zip)

**Comment by endrift** [comment]
FFA uses MBC2, which stores saves on a 512x4 bit chip. Most emulators expand this to 512x8 bits while leaving the top 4 bits blank. As this is a waste of space, I decided to do it differently. While it would be possible to convert this internally, mGBA does not do this at this time.

**Comment by RetroEdit** [comment]
Here's the thing: the explanation about mGBA using a more compressed format makes sense in theory, but isn't how it actually worked in this case. Formatting-wise, mGBA's save file for FFA looks identical to the original VBA save.

With that in mind, I gave it a further look. Every emulator does something completely different. For VBA, specifically all the top nibbles are zeroed like you mention, but it's still only 256 bytes instead of 512 bytes. Looking at when I opened it in Sameboy, it padded out the remaining bytes with `0xFF` up to 512 bytes. BGB has an odd repeating pattern of 8 random-looking bytes, followed by 8 `0xFF` up to 512 bytes. I also tested Gambatte (r571), and it padded the save file with `0xFF` up to 8192 bytes. I have attached the save files to this issue for future reference.

[FFA_save_file_comparison.zip](https://github.com/mgba-emu/mgba/files/4549037/FFA_save_file_comparison.zip)
